packages/watch-eth/test/utils.ts

```diff
@@ -0,0 +1,53 @@
+import { getLogger, sleep } from '@pigi/core'
+import { ethers } from 'ethers'
+import { EthereumListener } from '../src/interfaces/listener'
+import * as assert from 'assert'
+
+const log = getLogger('watch-eth-test-utils', true)
+
+export class TestListener<T> implements EthereumListener<T> {
+  private received: T[]
+
+  public constructor(private readonly sleepMillis = 50) {
+    this.received = []
+  }
+
+  public async handle(t: T): Promise<void> {
+    log.debug(`Received ${JSON.stringify(t)}`)
+    this.received.push(t)
+  }
+
+  public getReceived(): T[] {
+    return this.received.splice(0)
+  }
+
+  public async waitForReceive(num: number = 1): Promise<T[]> {
```

Comment:
Ah ok was confused about this at first, but to make sure I understand: this is a test helper which allows us to wait until we've received all of the messages which we expect & then returns. This will hang if we don't get all the messages but that's ok because the test will fail because of the timeout.

Reply:
Yep! 😄